src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -83,11 +57,92 @@ private boolean isInsertable(final Section section, final Section newSection) {
                 && !Objects.equals(newSection.getDownStation(), newSection.getUpStation());
     }
 
-    public List<StationResponse> toResponse() {
-        return sections.stream()
-                .flatMap(o -> o.getStations().stream())
-                .distinct()
-                .map(o -> new StationResponse(o.getId(), o.getName(), o.getCreatedDate(), o.getModifiedDate()))
+    public List<Section> getSections() {
+        return sections;
+    }
+
+    public void deleteByStationId(final Long stationId) {
+        List<Section> sections = findSectionsByStationId(stationId);
+        validateDeletableSection(sections);
+        delete(sections);
+    }
+
+    private void delete(final List<Section> sections) {
+        validateWrongSections(sections);
+        if (sections.size() == COUNT_OF_ENDPOINT_SECTION) {
+            removeEndPoint(sections);
+            return;
+        }
+        connect(sections);
+    }
+
+    private void validateWrongSections(final List<Section> sections) {
+        if (sections.size() > COUNT_OF_MIDPOINT_SECTION) {
+            throw new IllegalArgumentException("구간의 수가 올바르지 않습니다.");
+        }
+    }
+
+    private void validateDeletableSection(final List<Section> connectedSection) {
+        if (connectedSection.size() == NONE_SECTION || sections.size() == LAST_SECTION) {
+            throw new IllegalArgumentException("더 이상 삭제할 수 있는 구간이 없습니다.");
+        }
+    }
```

Comment:
`List` 의 메소드를 그대로 활용하셔도 명확하게 표현할 수 있을 것 같아요!!😆

```suggestion
    private void validateDeletableSection(final List<Section> connectedSection) {
        if (connectedSection.isEmpty()|| sections.isEmpty()) {
            throw new IllegalArgumentException("더 이상 삭제할 수 있는 구간이 없습니다.");
        }
    }
```